rholang/src/test/scala/coop/rchain/rholang/interpreter/matcher/MatchTest.scala

```diff
@@ -935,29 +937,49 @@ class VarMatcherSpec extends FlatSpec with Matchers with TimeLimits with TripleE
   }
 
   private def doMatchAndCharge(initialPhlo: Int) = {
-    implicit val cost: _cost[Task] = CostAccounting.unsafe[Task](Cost(initialPhlo))
-    val target: Par                = EList(Seq(GInt(1), GInt(2), GInt(3)))
+    implicit val cost: _cost[Task]                         = CostAccounting.unsafe[Task](Cost(initialPhlo))
+    implicit val costLog: FunctorListen[Task, Chain[Cost]] = costWriter[Task]
+    val target: Par                                        = EList(Seq(GInt(1), GInt(2), GInt(3)))
     val pattern: Par =
       EList(Seq(GInt(1), EVar(FreeVar(0)), EVar(FreeVar(1))), connectiveUsed = true)
 
-    (for {
+    val program = (for {
       res      <- spatialMatchAndCharge[Task](target, pattern).attempt
       phloLeft <- cost.get
-    } yield (phloLeft, res)).unsafeRunSync
+    } yield (phloLeft, res))
+
+    costLog.listen(program).unsafeRunSync
   }
 
   "spatialMatchAndCharge" should "short-circuit when runs out of phlo in the middle of matching" in {
-    val (_, res) = doMatchAndCharge(initialPhlo = 0)
+    val ((_, res), _) = doMatchAndCharge(initialPhlo = 0)
     res should be(Left(OutOfPhlogistonsError))
   }
 
   it should "charge for matching operations" in {
-    val (phloLeft, _) = doMatchAndCharge(initialPhlo = 100)
+    val ((phloLeft, _), log) = doMatchAndCharge(initialPhlo = 100)
     phloLeft.value shouldBe 90
+    log.toList should contain theSameElementsAs List(
+      Cost(4, "equality check"),
+      Cost(0, "comparison * 0"),
```

Comment:
we probably shouldn't ever allow a `Cost(operation = "")`. This way, this line would state e.g. 
`Cost(0, "comparison * number of patterns")`
or
`Cost(0, "comparison * number of sends")`

Reply:
The zeroes are worrying, but that's a different story